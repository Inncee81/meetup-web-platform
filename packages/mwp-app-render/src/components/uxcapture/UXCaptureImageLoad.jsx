// @flow
import * as React from 'react';

type Props = React$ElementConfig<HTMLImageElement> & { mark: string };

/**
 * Creates an image tag with provide props
 * and a UXCapture `onLoad` handler
 *
 * @see example https://github.com/meetup/ux-capture#image-elements
 */
const UXCaptureImageLoad = (props: Props) => {
	const { mark, src, ...other } = props;

	if (window && window.UX && window.UX[`${mark}-LOADED`]) {
		console.log(`${mark}-LOADED`);
		return <img src={src} {...other} />;
	}

	const onload = `
		if (window.UX) {
			window.UX.mark(${mark});
			console.log('MARKED: ${mark}');
		}
		window.UX.${mark}-LOADED = true;
	`;

	const onloadSingleLine = onload.replace(/[\n\t]+/g, ' ');

	return (
		<div
			dangerouslySetInnerHTML={{
				__html: `
				<img id="ux-capture-${mark}" src="${src}" onload="${onloadSingleLine}" />
			`,
			}}
		/>
	);
};

export default UXCaptureImageLoad;
