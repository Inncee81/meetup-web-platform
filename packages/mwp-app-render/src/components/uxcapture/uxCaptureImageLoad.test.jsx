import React from 'react';
import { shallow } from 'enzyme';
import UXCaptureImageLoad from './UXCaptureImageLoad';

const MOCK_PROPS = {
	mark: 'ux-image-load-logo',
	src: '/foo.png',
	className: 'someClass',
	alt: 'altText',
	height: '50',
};

const renderComponent = (props = MOCK_PROPS) =>
	shallow(<UXCaptureImageLoad {...props} />);

describe('UXCaptureImageLoad', () => {
	it('renders component markup', () => {
		global.window = { UX: { mark: jest.fn() } };
		expect(renderComponent()).toMatchSnapshot();
		delete global.window;
	});
});
