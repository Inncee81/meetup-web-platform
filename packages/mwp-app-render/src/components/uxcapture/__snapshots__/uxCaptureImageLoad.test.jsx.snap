// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UXCaptureImageLoad renders component markup 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "
				<img id=\\"ux-capture-ux-image-load-logo\\" src=\\"/foo.png\\" onload=\\" if (window.UX && !window.UX['ux-image-load-logo-LOADED']) { window.UX.mark('ux-image-load-logo'); window.UX['ux-image-load-logo-LOADED'] = true; } \\" className=\\"someClass\\" alt=\\"altText\\" height=\\"50\\" />
			",
    }
  }
/>
`;

exports[`getOnLoadJS renders the correct javascript as a string 1`] = `" if (window.UX && !window.UX['ux-image-load-logo-LOADED']) { window.UX.mark('ux-image-load-logo'); window.UX['ux-image-load-logo-LOADED'] = true; } "`;

exports[`getPropsAsHTMLAttrs renders the correct javascript as a string 1`] = `"className=\\"someClass\\" alt=\\"altText\\" height=\\"50\\""`;
